### PR TITLE
Implement QdrantExperiment

### PR DESCRIPTION
**Pull request by kacperlukawski**
This PR is an integration of Qdrant into prompttools. It's still in progress.

### CODE REVIEW

File: prompttools/experiment/experiments/qdrant_experiment.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+                for query_args in self.query_argument_combos:
+                    query_args = self._create_nested_object(query_args)
+                    for _ in range(runs):
+                        self.queue.enqueue(self.completion_fn, query_args)
+
+                self.results.extend(self.queue.results())
+                self.scores["latency"].extend(self.queue.latencies())
+                if len(self.results) == 0:
+                    logging.error("No results. Something went wrong.")
+                    raise PromptExperimentException
```
Comment (NivekT) [inline prompttools/experiment/experiments/qdrant_experiment.py:147]: To save the results, you want to call `_construct_result_dfs(...)` here. 

See this updated [source code](https://github.com/hegelai/prompttools/blob/main/prompttools/experiment/experiments/experiment.py#L178).

**Review by NivekT — COMMENTED** [review]
Hi,

Thank you for opening this PR. We just recently refactored how `Experiment` results are saved and how `experiment.evaluate()` work.

I added inline comments to walk through the necessary change. Please let me know if there is any question.

**Comment by kacperlukawski** [comment]
@HashemAlsaket @NivekT I implemented the requested changes.

**Comment by HashemAlsaket** [comment]
@kacperlukawski this is super cool. I just tested on my end and it works smoothly. Last few changes and I think it's good to go:

- Convert `examples/prompttests/test_qdrant.py` to a notebook example and save it here: `examples/notebooks/`
- Add instructions at the top of the notebook so users can run the example smoothly: [To run this example notebook, make sure you pull necessary Qdrant  setup using `docker pull qdrant/qdrant` and `docker run -p 6333:6333 qdrant/qdrant` from the Qdrant site (https://qdrant.tech/).
- Name the notebook similar to the others like `QdrantExperiment.ipynb`

**Review by NivekT — COMMENTED** [review]
Thanks for your work on this! Agree that a notebook example will be super useful for the users.

Also, please sign our CLA. Don't hesitate to let us know if you have any question!

**Comment by kacperlukawski** [comment]
CLA is signed. @NivekT Can we reopen the PR?

**Comment by kacperlukawski** [comment]
Thanks, I'll make the requested changes today!

**Comment by kacperlukawski** [comment]
@HashemAlsaket @NivekT Hope the changes are ready to be merged

**Review by NivekT — APPROVED** [review]
@kacperlukawski Thank you so much for your work!

I'm happy to merge it as it is, though if there is some way for `query_vector` to be rendered in a prettier way in the notebook dataframe, I'd be happy to explore.

Screenshot 2023-08-23 at 12 59 30 AM


If not, I will merge it later today. Thanks again!

**Review by NivekT — APPROVED** [review]

**Comment by NivekT** [comment]
Thank you so much @kacperlukawski

**Comment by HashemAlsaket** [comment]
Awesome stuff @kacperlukawski 🚀🚀🚀

**Pull request merged**